### PR TITLE
Fix NULL handling in the `SerpPreview` widget

### PR DESCRIPTION
**Pull request by aschempp**
Description (and possibily title) could be nullable fields in the database. I got the following error from calling `StringUtil::substr` on the value:

```
Deprecated: preg_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated
```

see https://terminal42.sentry.io/share/issue/fbb102c0dc534b939824576c09de1bc1/

### CODE REVIEW

**Review by Toflar — APPROVED** [review]

File: core-bundle/src/Resources/contao/widgets/SerpPreview.php [hunk 1]
```diff
@@ -109,20 +109,20 @@ private function getTitle(Model $model)
 	{
 		if (!isset($this->titleFields))
 		{
-			return $model->title;
+			return (string) $model->title;
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/widgets/SerpPreview.php:112]: The description is nullable indeed, however, the title is not.

https://github.com/contao/contao/blob/8114301bc2ef7bedd90dcc5afc9ee634cd526dd2/core-bundle/src/Resources/contao/dca/tl_page.php#L229

So these changes have to be reverted.

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/widgets/SerpPreview.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/widgets/SerpPreview.php:112]: well not for `tl_page`. But you don't know all the tables SerpWidget is used on!

Comment (leofeyer) [reply core-bundle/src/Resources/contao/widgets/SerpPreview.php:112]: <del>This is a pointless discussion. The code inspector would detect and remove the unnecessary type cast anyway.</del>

I see the point.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/widgets/SerpPreview.php:112]: IMHO we should fix this in lines 44 and 45:

```php
$title = StringUtil::substr((string) $this->getTitle($model), 64);
$description = StringUtil::substr((string) $this->getDescription($model), 160);
```

Comment (aschempp) [reply core-bundle/src/Resources/contao/widgets/SerpPreview.php:112]: well I was assuming the `getTitle()` and `getDescription()` should have typed return values at some point. And then they would need to cast the value.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/widgets/SerpPreview.php:112]: But even then, the return type would most likely be `string|null`, so one can distinguish between "not set" and "empty".

Comment (aschempp) [reply core-bundle/src/Resources/contao/widgets/SerpPreview.php:112]: I don't really care 🤷

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**